src/components/Admin/DropDown.jsx

```diff
@@ -1,59 +1,76 @@
 import React, { useState } from "react";
 import Dropdown from "react-bootstrap/Dropdown";
 import { RiArrowDownSLine } from "react-icons/ri";
-const Toggle = ({ children, onClick }) => {
+
+const Toggle = ({ option, onClick, show }) => {
   return (
     <button
       onClick={onClick}
       className="relative z-50 bg-white text-black w-full border-2 rounded-full border-black justify-between flex items-center px-3 py-1"
     >
-      {children}
-      <RiArrowDownSLine />
+      {option}
+      <RiArrowDownSLine className={`${show && "rotate-180"} duration-300`} />
     </button>
   );
 };
-const Menu = ({ children, className, setOptions, options }) => {
+const Menu = ({ setOption, className, setOptions, options }) => {
+  const handleInput = (e) => {
+    setValue(e.target.value);
+    setOptions(
+      options.map((option) => ({
+        ...option,
+        hidden: !option.name
+          .toLowerCase()
+          .includes(e.target.value.toLowerCase()),
+      }))
+    );
+  };
   const [value, setValue] = useState("");
   return (
     <div className={className}>
       <input
         autoFocus
-        className="mx-3 my-2 w-fit ring-0 outline-none"
+        className="mx-3 my-2 w-11/12 ring-0 outline-none border-gray-300 border-[1px] text-sm px-2 py-1"
         placeholder="Type to filter..."
-        onChange={(e) => {
-          setValue(e.target.value);
-          setOptions(
-            options.filter(
-              (option) => !value || option.toLowerCase().includes(value)
-            )
-          );
-        }}
+        onChange={handleInput}
         value={value}
       />
-      {children}
+      {options.filter((opt) => !opt.hidden).length > 0 ? (
+        options.map(
+          (option, index) =>
+            !option.hidden && (
```

Comment:
filter the array first and then map through it, it will remove the need for this check